src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +165,999 @@ static DamageTable D_80A63CE0 = {
+    /* 12 */ { &gTingleHideFaceAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
+    /* 13 */ { &gTingleThrowConfettiAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
 };
 
-#endif
+static Vec3f sInitBalloonScale = { 0.0f, 0.0f, 0.0f };
+
+void EnBal_Init(Actor* thisx, PlayState* play) {
+    EnBal* this = THIS;
+    s32 pad;
+    f32 lastFrame;
+
+    lastFrame = Animation_GetLastFrame(&gTingleFloatIdleAnim);
+    this->location = EN_BAL_GET_LOCATION(&this->picto.actor);
+    this->picto.actor.targetMode = 1;
+    this->picto.actor.uncullZoneForward = 3000.0f;
+    Actor_SetScale(&this->picto.actor, 0.02f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gTingleSkel, &gTingleFloatIdleAnim, this->jointTable, this->morphTable,
+                       TINGLE_LIMB_MAX);
+    if (gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
+        Animation_Change(&this->skelAnime, &gTingleTalkIdleAnim, 1.0f, 0.0f, lastFrame, ANIMMODE_LOOP, -10.0f);
+    }
+    ActorShape_Init(&this->picto.actor.shape, 0.0f, ActorShadow_DrawCircle, 0.0f);
+    this->skyFloatPhase = 0;
+    this->lastMessageId = 0;
+    this->lastBalloonAction = EN_BAL_BALACT_NONE;
+    this->timer = 0;
+    this->watchTarget = EN_BAL_WATCHTRG_NONE;
+    this->inflateEarly = false;
+    this->isTalking = false;
+    this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+    this->eyeTimer = 60;
+    this->forceEyesShut = false;
+    Collider_InitJntSph(play, &this->collider);
+    Collider_SetJntSph(play, &this->collider, &this->picto.actor, &sJntSphInit, &this->colliderElement);
+    CollisionCheck_SetInfo2(&this->picto.actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    Actor_UpdateBgCheckInfo(play, &this->picto.actor, 0.0f, 0.0f, 0.0f, UPDBGCHECKINFO_FLAG_4);
+    this->picto.validationFunc = EnBal_ValidatePictograph;
+    if (!gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
+        this->picto.actor.world.pos.y = this->picto.actor.floorHeight;
+        EnBal_SetMainColliderToHead(this);
+        EnBal_SetupGroundIdle(this);
+    } else {
+        EnBal_SetupFloatIdle(this);
+    }
+}
+
+void EnBal_Destroy(Actor* thisx, PlayState* play) {
+    EnBal* this = THIS;
+    Collider_InitJntSph(play, &this->collider);
+}
+
+void EnBal_SetMainColliderToBalloon(EnBal* this) {
+    this->collider.elements->dim.limb = TINGLE_LIMB_BALLOON;
+    this->collider.elements->dim.modelSphere.radius = 40;
+    this->collider.elements->dim.modelSphere.center.x = 2200;
+}
+
+void EnBal_SetMainColliderToHead(EnBal* this) {
+    this->collider.elements->dim.limb = TINGLE_LIMB_HEAD;
+    this->collider.elements->dim.modelSphere.radius = 25;
+    this->collider.elements->dim.modelSphere.center.x = 0;
+}
+
+s32 EnBal_ValidatePictograph(PlayState* play, Actor* thisx) {
+    s32 pictoValid;
+    EnBal* this = THIS;
+
+    pictoValid = Snap_ValidatePictograph(play, thisx, PICTO_VALID_TINGLE, &this->picto.actor.focus.pos,
+                                         &thisx->shape.rot, 10.0f, 400.0f, 0x4000);
```

Comment:
```suggestion
    pictoValid = Snap_ValidatePictograph(play, &this->pictor.actor, PICTO_VALID_TINGLE, &this->picto.actor.focus.pos,
                                         &this->picto.actor.shape.rot, 10.0f, 400.0f, 0x4000);
```
Does this work? if not can you change `&this->picto.actor.foucs.pos` to `&thisx->focus.pos` so they all use `thisx`?